src/utils/utils.ts

```diff
@@ -1,10 +1,10 @@
 import { Dispatch, SetStateAction } from "react";
-import { querySearchService } from "../api/browse";
+import { querySearchService, urlWithEndSlash } from "../api/browse";
```

Comment:
in order to resolve the circular import, I would have moved urlWithEndSlash to here since it's clearly a common utility function, and handleFilterAndSearch away from here to browse because it's not a common utility but specific to the browse module

but let's keep it like this for now, we can refactor later